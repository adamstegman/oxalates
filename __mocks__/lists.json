[
  {
    "id": "all",
    "name": "All",
    "bottomThreshold": null,
    "topThreshold": null
  },
  {
    "id": 1,
    "name": "Very High",
    "bottomThreshold": 50,
    "topThreshold": null
  },
  {
    "id": 2,
    "name": "High",
    "bottomThreshold": 15,
    "topThreshold": 50
  },
  {
    "id": 3,
    "name": "Moderate",
    "bottomThreshold": 5,
    "topThreshold": 15
  },
  {
    "id": 4,
    "name": "Low",
    "bottomThreshold": null,
    "topThreshold": 5
  }
]
