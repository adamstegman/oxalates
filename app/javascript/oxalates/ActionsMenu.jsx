import React from 'react';

import { ActiveSessionMenu } from './ActiveSessionMenu';
import { FoodListSearchForm } from './FoodListSearchForm';

import './ActionsMenu.scss';

export class ActionsMenu extends React.Component {
  render() {
    return (
      <div className="actions">
        <ActiveSessionMenu />
        <FoodListSearchForm />
      </div>
    );
  }
};

ActionsMenu.propTypes = {
};
