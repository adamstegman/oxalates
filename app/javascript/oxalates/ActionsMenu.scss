@import "./constants";

$actions_height: 44px;
$actions_background_color: lighten($title_background_color, 20%);
$add_food_padding: $action_padding;
$add_food_width: 2em;
$edit_food_padding: $action_padding;
$edit_food_width: 2em;

.actions {
  display: -webkit-flex;
  -webkit-align-items: center;
  -webkit-justify-content: space-between;
  display: flex;
  align-items: center;
  justify-content: space-between;

  min-height: $actions_height;

  background-color: $actions_background_color;
  font-size: $actions_font_size;
  a {
    text-decoration: none;
  }

  .add-food {
    padding: $add_food_padding;
    min-width: $add_food_width;

    text-align: center;
  }

  .edit-food {
    padding: $edit_food_padding;
    min-width: $edit_food_width;

    text-align: center;
  }
}
