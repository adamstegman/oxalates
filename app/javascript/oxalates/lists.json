[
  {
    "id": "all",
    "name": "All",
    "bottom_threshold": null,
    "top_threshold": null
  },
  {
    "id": 1,
    "name": "Very High",
    "bottom_threshold": 50,
    "top_threshold": null
  },
  {
    "id": 2,
    "name": "High",
    "bottom_threshold": 15,
    "top_threshold": 50
  },
  {
    "id": 3,
    "name": "Moderate",
    "bottom_threshold": 5,
    "top_threshold": 15
  },
  {
    "id": 4,
    "name": "Low",
    "bottom_threshold": null,
    "top_threshold": 5
  }
]
