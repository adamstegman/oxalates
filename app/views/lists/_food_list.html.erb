<ul class="food-list">
<% foods.each do |food| %>
  <li class="food-list-item <%= list_class_name(food.list) %>"><%= food.name %></li>
<% end %>
</ul>
