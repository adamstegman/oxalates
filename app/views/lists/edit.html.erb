<%= form_tag list_path(@list.id), method: :put do %>
  <ul class="food-list">
  <% @list.foods.each do |food| %>
    <li>
      <%= link_to food.name, edit_food_path(food), class: "edit-food-list-item #{list_class_name(food.list)}" %>
      <%= hidden_field_tag 'list[foods][][id]', food.id %>
      <%= label_tag 'list[foods][][name]', "Name", class: 'food-list-item-name' %>
      <%= text_field_tag 'list[foods][][name]', food.name, class: 'food-list-item-name' %>
    </li>
  <% end %>
  </ul>
<% end %>
