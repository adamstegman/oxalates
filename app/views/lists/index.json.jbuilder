json.lists(@lists) do |list|
  json.(list, :id, :name, :bottom_threshold, :top_threshold)
end
