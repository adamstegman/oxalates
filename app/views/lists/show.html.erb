<%= render "food_list", foods: @list.foods %>
