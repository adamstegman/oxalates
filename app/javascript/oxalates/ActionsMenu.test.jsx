import React from 'react';
import renderer from 'react-test-renderer';
import { Provider } from 'react-redux';
import configureStore from 'redux-mock-store';

import lists from './__mocks__/lists.json';
import { ActionsMenu } from './ActionsMenu';

const mockStore = configureStore();
const state = {
  session: {
    authenticated: true,
    authenticating: false,
    password: 'password',
  },
  foodList: {
    query: 'test',
  },
  listMenu: {
    activeListId: lists[0].id,
  },
};

test('ActionsMenu renders the actions menu', () => {
  const store = mockStore(state);
  const component = renderer.create(
    <Provider store={store}>
      <ActionsMenu />
    </Provider>,
  );
  let tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});
