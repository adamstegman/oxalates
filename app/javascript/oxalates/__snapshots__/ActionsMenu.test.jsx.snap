// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ActionsMenu renders the actions menu 1`] = `
<div
  className="actions"
>
  <button
    className="action logout"
    onClick={[Function]}
    type="button"
  >
    Log out
  </button>
  <form
    className="search"
    onSubmit={[Function]}
  >
    <label
      htmlFor="search"
    >
      Search
    </label>
    <input
      name="search"
      onChange={[Function]}
      placeholder="Search"
      type="search"
      value="test"
    />
    <input
      type="submit"
      value="Search"
    />
  </form>
</div>
`;
